src/physics.cpp

```diff
@@ -1162,5 +1175,97 @@ void sample_secondary_photons(Particle& p, int i_nuclide)
     p.create_secondary(u, E, Particle::Type::photon);
   }
 }
+	
+void split_particle(Particle& p)
+{
+  // weight window  add by Yuan
+
+  std::vector<double> energy_group;
+  std::vector<double> ww_lower;
+  double lower_ww;       
+  double upper_ww;
+  double survival_ww;
+  int    max_split;
+  // Determine which set of weight window values to be used based on particle type
+  if (p.type_==Particle::Type::neutron) {
+    energy_group = settings::n_energy_group;
+    ww_lower     = settings::n_ww_lower;
+    lower_ww     = settings::n_multiplier;       
+    upper_ww     = settings::n_upper_ratio;
+    survival_ww  = settings::n_survival_ratio;
+    max_split    = settings::n_max_split;
+  } else if (p.type_==Particle::Type::photon) {
+    energy_group = settings::p_energy_group;
+    ww_lower     = settings::p_ww_lower;
+    lower_ww     = settings::p_multiplier;       
+    upper_ww     = settings::p_upper_ratio;
+    survival_ww  = settings::p_survival_ratio;
+    max_split    = settings::p_max_split;
+  }
+
+  // Particle's position, weight and energy
+  Position pos  = p.r();
+  double weight = p.wgt_;
+  double Energy = p.E_;	
+
+  // Check if this particle is in the weight weindow mesh
+  for (int i = 0; i < 3; ++i) {
+    if ((pos[i] < settings::lower_left_point[i]) || (pos[i] > settings::upper_right_point[i])) return;
+  }
+	
+  // index for position and energy
+  int ijk[3]     = {0}; // mesh bin in each direction
+  int energy_bin = 0;   // energy bin
+  int indices    = 0;   // indices in weight window vector
+	
+  // get the mesh bin in x direction
+  auto i = lower_bound_index(settings::mesh_x.begin(), settings::mesh_x.end(), pos[0]);
+  ijk[0] = i + 1;
```

Comment:
It still looks to me like we could avoid incrementing the indices here and then decrementing them later...

Reply:
You are right, I just noticed this.